File: Year-2023/LeetCode-May-Challenge/MaximumNumberofVowelsinaSubstringofGivenLength.js
var maxVowels = function (s, k) {
    let countVowels = 0, tempCount = 0;
    for (let i = 0; i < s.length; ++i) {
        if (s[i] === 'a' || s[i] === 'e' || s[i] === 'i' || s[i] === 'o' || s[i] === 'u') tempCount++;
        if (i - k >= 0)
            if (s[i - k] === 'a' || s[i - k] === 'e' || s[i - k] == 'i' || s[i - k] == 'o' || s[i - k] == 'u') tempCount--;
        if (tempCount > countVowels) countVowels = tempCount;
    }
    return countVowels;
};

let s = "abciiidef", k = 3;
console.log(maxVowels(s, k));

File: Year-2023/LeetCode-May-Challenge/MaximumNumberofVowelsinaSubstringofGivenLength.java
class Solution {
    public int maxVowels(String s, int k) {
        int countVowels = 0, tempCount = 0;
        for(int i = 0; i < s.length(); ++i){
            if(s.charAt(i) == 'a' || s.charAt(i) == 'e' || s.charAt(i) == 'i' || s.charAt(i) == 'o' || s.charAt(i) == 'u') tempCount++;
            if(i - k >= 0)
                if(s.charAt(i - k) == 'a' || s.charAt(i - k) == 'e' || s.charAt(i - k) == 'i' || s.charAt(i - k) == 'o' || s.charAt(i - k) == 'u') tempCount--;
            if(tempCount > countVowels){
                countVowels = tempCount;
            }
        }
        return countVowels;
    }
}
public class MaximumNumberofVowelsinaSubstringofGivenLength{
    public static void main(String[] args) {
        String s = "abciiidef"; int k = 3;
        System.out.println(new Solution().maxVowels(s, k));
    }
}